File: Exercices/Séance 1 - C# vs Java-20231230/Ex1/Director.cs
namespace be.ipl.domaine;

// implements Serializable 
// [] -> annotations
[Serializable]
    internal class Director : Person
    {
	private List<Movie> _directedMovies;

	public Director(string name, string firstname, DateTime birthDate) : base (name, firstname, birthDate)
	{
		_directedMovies = new List<Movie>();
	}

	public bool AddMovie(Movie movie)
	{

		if (_directedMovies.Contains(movie))
			return false;

		if (movie.Director == null)
			movie.Director = this;

		_directedMovies.Add(movie);

		return true;

	}

	public IEnumerator<Movie> Movies()
	{
		return _directedMovies.GetEnumerator();
	}


}

File: Exercices/Séance 1 - C# vs Java-20231230/ActorsDB/src/be/ipl/domaine/Director.java
package be.ipl.domaine;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Iterator;
import java.util.List;

public class Director extends Person implements Serializable {

	private static final long serialVersionUID = 5952964360274024205L;
	private List<Movie> directedMovies;
	
	public Director(String name, String firstname, Calendar birthDate) {
		super(name, firstname, birthDate);
		directedMovies = new ArrayList<Movie>();
	}

	public boolean addMovie(Movie movie){
		
		if (directedMovies.contains(movie))
			return false;
		
		if (movie.getDirector() == null)
			movie.setDirector(this);
		
		directedMovies.add(movie);

		return true;
		
	}
	
	public Iterator<Movie> Movies(){
		return directedMovies.iterator();
	}
	
	
}
